sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -127,10 +127,12 @@ public CompletableResultCode forceFlush() {
     private long nextExportTime;
 
     private final BlockingQueue<ReadableSpan> queue;
```

Comment:
Using ArrayDequeue would be better when draining in batches similar to https://github.com/sbandadd/opentelemetry-java/pull/1/files. But I think the most important thing is to come up with a CPU benchmark and compare different versions.